pkgs/development/tools/richgo/default.nix

```diff
@@ -13,10 +13,12 @@ buildGoModule rec {
 
   vendorSha256 = "sha256-O63QEo0/+m9cYktMg4+RloLuUfAlCG0eGkxpHPFg/Cw=";
 
-  doCheck = false;
-
   subPackages = [ "." ];
 
+  postCheck = ''
+    go test -v --race ./...
+  '';
```

Comment:
Rather than re-defining go test with none of the extra flags handled by buildGoModule I do this

```suggestion
  # Feed in all tests for testing
  # This is because subPackages above limits what is built to just what we
  # want but also limits the tests
  preCheck = ''
    unset subPackages
  '';
```

maybe double check if `subPackages = [ "." ];` is a good idea or if `excludePackages` would be more appropriate

Reply:
I have removed everything and tests are running! Thanks for ideas!

Reply:
OK, awesome.

Did you double check the output didnt have extra nonsense binaries? Sometimes subpackages is used to avoid extra main.go files being built into binaries e.g. some go utility for updating their docs or something?

Reply:
Very clean.

![image](https://user-images.githubusercontent.com/5861043/164307002-31ca3ff1-6a9a-4d32-9b4f-6cd351cccf02.png)

Reply:
OK, great 👍